core/trino-web-ui/src/main/resources/webapp-preview/src/components/flow/StepNode.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Box, Card, CardContent, Grid2 as Grid, Tooltip, Typography } from '@mui/material'
+import { Handle, Position } from '@xyflow/react'
+import { STEP_NODE_WIDTH } from './layout'
+import { truncateString } from '../../utils/utils.ts'
+export interface IStepNodeProps {
+    data: {
+        label: string
+        descriptor: Map<string, string>
+    }
+}
+
+export const StepNode = (props: IStepNodeProps) => {
```

Comment:
Could you add a comment explaining what is a `StepNode` ?
Is this meant to be a leaf node ?

Reply:
sure, added